lib/src/DotenvParser.cc

```diff
@@ -0,0 +1,423 @@
+{
+    std::string ss = s;
+    ss.erase(s.find_last_not_of(c) + 1);
+    ss.erase(0, s.find_first_not_of(c));
+    return ss;
+}
+std::vector<std::string> splitBy(const std::string &str, const char dlm)
+{
+    std::vector<std::string> parts;
+    std::stringstream stream(str);
+    std::string part;
+    while (std::getline(stream, part, dlm))
+    {
+        parts.push_back(trim(part, " \"'"));
+    }
+    return parts;
+}
+std::string parseNamespace(const std::string &line)
+{
+    std::string ns;
+    for (const auto &c : line)
+    {
+        if (c == '[')
+        {
+            continue;
+        }
+        if (c == ']')
+        {
+            break;
+        }
+        ns += c;
+    }
+    return ns;
+}
+std::string removeInlineComments(const std::string &value)
+{
+    return splitBy(value, '#')[0];
+}
+std::pair<std::string, std::string> parseKeyValue(const std::string &line)
+{
+    auto parts = splitBy(line, '=');
+    if (parts.size() < 2)
+    {
+        return {"", ""};
+    }
+    return {parts[0], removeInlineComments(parts[1])};
+}
+bool isEqual(const std::string &a, const std::string &b)
+{
+    std::string upperA = a;
+    std::string upperB = b;
+    std::transform(upperA.begin(), upperA.end(), upperA.begin(), ::toupper);
+    std::transform(upperB.begin(), upperB.end(), upperB.begin(), ::toupper);
+    return upperA == upperB;
+}
+void DotenvParser::parse(std::ifstream &infile)
+{
+    std::ostringstream content;
+    if (infile.is_open())
+    {
+        content << infile.rdbuf();
+    }
+    infile.close();
+
+    std::vector<std::string> lines = splitBy(content.str(), '\n');
+    std::string ns;
+
+    int listenersIndex = -1;
+    int dbClientsIndex = -1;
+    int redisClientsIndex = -1;
+
+    for (const auto &line : lines)
+    {
+        if (line.empty())
+        {
+            continue;
+        }
+        std::string cleanLine = trim(line, " ");
+        if (cleanLine[0] == '#')
+        {
+            continue;
+        }
+        if (cleanLine[0] == '[')
+        {
+            ns = parseNamespace(cleanLine);
+            if (isEqual(ns, "listeners"))
+            {
+                listenersIndex++;
+            }
+            else if (isEqual(ns, "db_clients"))
+            {
+                dbClientsIndex++;
+            }
+            else if (isEqual(ns, "redis_clients"))
+            {
+                redisClientsIndex++;
+            }
+            continue;
+        }
+        auto pair = parseKeyValue(cleanLine);
+        if (isEqual(ns, "app"))
+        {
+            if (isEqual(pair.first, "number_of_threads"))
+            {
+                configJsonValue_["app"]["number_of_threads"] = std::stoi(pair.second);
+            }
+            else if (isEqual(pair.first, "enable_session"))
+            {
+                configJsonValue_["app"]["enable_session"] = isEqual(pair.second, "true");
+            }
+            else if (isEqual(pair.first, "session_timeout"))
+            {
```

Comment:
Is there a more modular way to do this? I feel maintaining this is going to be hard...

Reply:
Unfortunately, since `.env` doesn't support data types I can't think of a better way to do it, but, please take a look at my comment below about assigning the variables to the environment variables and manually using that in the app.